### DISCUSSION TITLE
how to make  a  dataset for training and testing  yolov5  model?

### DISCUSSION THREAD

**Discussion by kameronren** [deepjavalibrary/djl#2676] [Q&A]
how to make  a  dataset for training and testing  yolov5  model?

**Comment by kameronren** [comment]
anyone can help me?

**Comment by zachgk** [comment]
Take a look at our [Dataset Creation Guide](http://docs.djl.ai/docs/development/how_to_use_dataset.html) and you will want to extend the `ObjectDetectionDataset`.

You can then either separate the raw data into training and testing (if whatever data source you are using does not already split it). Otherwise, you can split it on the DJL end by using `dataset.randomSplit(...)` which is defined on `RandomAccessDataset`.
